src/Common/Storage/Timed_Option.php

```diff
@@ -0,0 +1,351 @@
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function deactivate(): void {
+		$this->active = false;
+	}
+
+	/**
+	 * Activate the usage of Database Timed Options.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function activate(): void {
+		$this->active = true;
+	}
+
+	/**
+	 * Is the timed options active?
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public function is_active(): bool {
+		/**
+		 * Allows the modification of the state of usage for Timed Options.
+		 *
+		 * @since TBD
+		 *
+		 * @param bool $active Whether we use Database Timed Options or a glorified Memoization system.
+		 */
+		return (bool) apply_filters( 'tec_common_timed_option_is_active', $this->active );
+	}
+
+	/**
+	 * Gets the option name for a given timed option, by attaching a prefix and allowing filtering.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $key Key for the option we are trying to get the option name for.
+	 *
+	 * @return string
+	 */
+	public function get_option_name( string $key ): string {
+		/**
+		 * Allows the modification of where we store the Transient Data.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $option_name Name of the option where all the transient data will live.
+		 */
+		return (string) apply_filters( 'tec_common_timed_option_name', $this->option_name_prefix . $key, $this->option_name_prefix );
+	}
+
+	/**
+	 * Fetches the value of a given timed option.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $key     Key for the option we are trying to get.
+	 * @param mixed  $default Default value when the option is either expired or not-set.
+	 * @param bool   $force   If we should expire cache and fetch from the database.
+	 *
+	 * @return mixed|null
+	 */
+	public function get( $key, $default = null, bool $force = false ) {
+		/**
+		 * Allows the filtering the default timed_option value.
+		 *
+		 * @since TBD
+		 *
+		 * @param mixed  $default Default value when the option is either expired or not-set.
+		 * @param string $key     Key for the option we are trying to get.
+		 * @param bool   $force   If we should expire cache and fetch from the database.
+		 */
+		$default = apply_filters( 'tec_common_timed_option_default_value', $default, $key, $force );
+
+		/**
+		 * Allows the filtering to short-circuit the whole fetch logic.
+		 *
+		 * @since TBD
+		 *
+		 * @param mixed|null $pre     If anything diff than null it will short-circuit.
+		 * @param string     $key     Key for the option we are trying to get.
+		 * @param mixed      $default Default value when the option is either expired or not-set.
+		 * @param bool       $force   If we should expire cache and fetch from the database.
+		 */
+		$pre = apply_filters( 'tec_common_timed_option_pre_value', null, $key, $default, $force );
+
+		if ( null !== $pre ) {
+			return $pre;
+		}
+
+		$time = time();
+
+		// If we have a stored value that is not expired, use it.
+		if (
+			! $force &&
```

Comment:
check on the placement of the && in the conditions here and other places. In some cases it is at the end of the line and some at the start.